### PR TITLE
remove `[new` from docs

### PR DESCRIPTION
**Pull request by dnfield**
Missed these in the PR where I got rid of optional news.

### CODE REVIEW

**Comment by a14n** [comment]
I'm not such how dartdoc handles comment references (as dartdoc and navigation don't always have the same behaviour - see dart-lang/sdk#36974) but in code navigation `[new Aaa]` brings to `Aaa` constructor and `[Aaa]` brings you to the class. So I'm not sure the result is what you want.

**Comment by dnfield** [comment]
For named constructors you definitely don't need the new - I've seen that working.  For default constructors I'm not sure what's better. I could take this out for defaults.

/cc @gspencergoog who might know more on this

**Comment by gspencergoog** [comment]
I think this is fine.  In IntelliJ at least if you Ctrl-click on `[new Size]`, it takes you to the `Size` class declaration anyhow, not the default constructor.  And named constructors work properly too by taking you to the named constructor and not the class declaration.  Dartdoc has different behavior: a link to `[new Size]` takes you to the default constructor page, and `[Size]` takes you to the class page.

We actually use that difference in a few places (e.g. see [TextFormField](https://master-api.flutter.dev/flutter/material/TextFormField/TextFormField.html) where it says "For documentation about the various parameters, see the TextField class and new TextField, the constructor.")

I'm not sure the distinction is all that relevant most times, but when it is, it's important to go where the user thinks they're going. I'd just make sure to look at each instance of removing the `new` and see if it makes sense for that case or not.

**Comment by dnfield** [comment]
Updated to prevserve new where it seems to make sense.

File: lib/ui/geometry.dart [hunk 1]
```diff
@@ -1689,7 +1689,7 @@ class RSTransform {
   /// to factor out the computations of the sine and cosine of the rotation
   /// (which are computed each time this constructor is called) and reuse them
   /// over multiple [RSTransform] objects, it may be more efficient to directly
-  /// use the more direct [new RSTransform] constructor instead.
```
Comment (a14n) [inline lib/ui/geometry.dart]: I think it refers to constructor here

File: lib/ui/painting.dart [hunk 2]
```diff
@@ -3553,7 +3553,7 @@ class Canvas extends NativeFieldWrapperClass2 {
   ///
   /// To align the text, set the `textAlign` on the [ParagraphStyle] object
   /// passed to the [new ParagraphBuilder] constructor. For more details see
-  /// [TextAlign] and the discussion at [new ParagraphStyle].
```
Comment (a14n) [inline lib/ui/painting.dart]: The discussion mentioned seems to be on `new ParagraphStyle`

File: lib/ui/geometry.dart (hunk 1 above)
Comment (dnfield) [reply lib/ui/geometry.dart]: Fixed

File: lib/ui/text.dart [hunk 3]
```diff
@@ -1203,8 +1203,8 @@ class ParagraphConstraints {
   /// forced line break is placed after it (even if an explicit line break
   /// follows).
   ///
-  /// The width influences how ellipses are applied. See the discussion at [new
-  /// ParagraphStyle] for more details.
+  /// The width influences how ellipses are applied. See the discussion at
+  /// [ParagraphStyle] for more details.
```
Comment (a14n) [inline lib/ui/text.dart]: ditto

File: lib/ui/text.dart [hunk 4]
```diff
@@ -1346,8 +1346,8 @@ class Paragraph extends NativeFieldWrapperClass2 {
   /// null, `ellipsis` was not null, and one of the lines exceeded the width
   /// constraint.
   ///
-  /// See the discussion of the `maxLines` and `ellipsis` arguments at [new
-  /// ParagraphStyle].
+  /// See the discussion of the `maxLines` and `ellipsis` arguments at
+  /// [new ParagraphStyle].
```
Comment (a14n) [inline lib/ui/text.dart]: ditto

Comment (dnfield) [reply lib/ui/text.dart]: This one is ok, right?  I just moved new down to the next line.

Comment (a14n) [reply lib/ui/text.dart]: right. sorry

**Comment by jamesderlin** [comment]
BTW, also see https://github.com/dart-lang/dartdoc/issues/1850.

**Comment by Hixie** [comment]
> Dartdoc has different behavior: a link to `[new Size]` takes you to the default constructor page, and `[Size]` takes you to the class page.

Dartdoc is canonical.

Whatever we do, please make sure it works in dartdocs (i.e. that we can link to constructors when we need to), and that it is documented.

**Pull request merged**